cluster-autoscaler/provisioningrequest/checkcapacity/provisioningclass.go

```diff
@@ -93,12 +94,18 @@ func (o *checkCapacityProvClass) Provision(
 }
 
 // Assuming that all unschedulable pods comes from one ProvisioningRequest.
-func (o *checkCapacityProvClass) checkcapacity(unschedulablePods []*apiv1.Pod, provReq *provreqwrapper.ProvisioningRequest) (bool, error) {
+func (o *checkCapacityProvClass) checkcapacity(unschedulablePods []*apiv1.Pod, provReq *provreqwrapper.ProvisioningRequest) (capacityAvailable bool, err error) {
+	capacityAvailable = true
 	st, _, err := o.injector.TrySchedulePods(o.context.ClusterSnapshot, unschedulablePods, scheduling.ScheduleAnywhere, true)
 	if len(st) < len(unschedulablePods) || err != nil {
 		conditions.AddOrUpdateCondition(provReq, v1beta1.Provisioned, metav1.ConditionFalse, conditions.CapacityIsNotFoundReason, "Capacity is not found, CA will try to find it later.", metav1.Now())
-		return false, err
+		capacityAvailable = false
+	} else {
+		conditions.AddOrUpdateCondition(provReq, v1beta1.Provisioned, metav1.ConditionTrue, conditions.CapacityIsFoundReason, conditions.CapacityIsFoundMsg, metav1.Now())
 	}
-	conditions.AddOrUpdateCondition(provReq, v1beta1.Provisioned, metav1.ConditionTrue, conditions.CapacityIsFoundReason, conditions.CapacityIsFoundMsg, metav1.Now())
-	return true, nil
+	_, updErr := o.client.UpdateProvisioningRequest(provReq.ProvisioningRequest)
+	if updErr != nil {
+		klog.Errorf("failed to update Provisioned condition to ProvReq %s/%s, err: %v", provReq.Namespace, provReq.Name, err)
```

Comment:
Any reason we don't return `updErr` when `err == nil`? Even if the capacity is there, we have failed to let the workload use it. Given that these errors are mostly used for observability, I'd rather give the user a signal things are not working than not.

The only reason I can think of to avoid it is that it might cause liveness probe to restart CA after some time if it has nothing else to do (no unschedulable pods in the cluster to process) and keeps failing to update a ProvisioningRequest every iteration. But if it has nothing else to do anyway, it's probably not a big deal.

Reply:
+1 We also, we definitely should log `updErr`

Reply:
Thanks Alex, for catching this!